src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/GenerateDocumentHandlerTest.java

```diff
@@ -393,4 +419,68 @@ void should_not_allow_null_arguments() {
             .hasMessage("callback must not be null")
             .isExactlyInstanceOf(NullPointerException.class);
     }
+
+    @Test
+    void should_send_decision_and_reasons_and_populate_ftpa_field_ada_appeal() {
+        AsylumCase expectedUpdatedCase = mock(AsylumCase.class);
+        setUpSendDecisionsAndReasonsData(expectedUpdatedCase);
+
+        when(expectedUpdatedCase.read(IS_ACCELERATED_DETAINED_APPEAL, YesOrNo.class))
+                .thenReturn(Optional.of(YesOrNo.YES));
+
+        final DueDateService dueDateServiceObject = new DueDateService(new HolidayService(holidays));
+        final ZonedDateTime fakeAppealDateTime = FAKE_APPEAL_DATE.atStartOfDay(ZoneOffset.UTC);
+        final ZonedDateTime actualFtpaDeadline = dueDateServiceObject.calculateDueDate(fakeAppealDateTime, FTPA_DUE_IN_WORKING_DAYS_ADA);
```

Comment:
Suggestion: You dont need to create another DueDateService object to calculate the deadline date. You can calculate based on calendar and make it less dynamic as well.